routes/story.route.server.js

```diff
@@ -80,7 +81,79 @@ module.exports = app => {
                 .then(story => res.json(story))
                 .catch((error) => res.status(500).send({error}));
     };
-  
+
+    // there is no check for userId being valid here. The expectation is that only validated users would be able to
+    // navigate to comment page
+    let addComment = (req,res) => {
+
+        const storyId = req.body.storyId;
+        const userId = req.body.userId;
+        const content = req.body.content;
+        const date = req.body.date;
+
+        storyDao.findStoryByStoryID(storyId).then(story => {
+            if(story){
+                commentDao.addComment(userId,content,date).then(comment => {
+                    story.comments.push(comment);
+                    storyDao.updateStory(story.story_id,story).then(updatedStory => {
+                        res.send(updatedStory)
+                    })
+                });
+            }
+            else{
+                res.status(403).send({
+                    success: false,
+                    message: "Story does not exist or has been deleted."
+                })
+            }
+        })
+    };
+
+    // Only allows users to delete their own comment
+    //Todo: Modify to allow moderators or admin to delete as well.
+    let deleteComment = (req,res) => {
+        const storyId = req.body.storyId;
+        const userId = req.body.userId;
+        const commentId = req.body.commentId;
+
+        storyDao.findStoryByStoryID(storyId).then(story => {
+            if(story){
+                commentDao.findCommentById(commentId).then(comment => {
+                    if(comment){
+                        if(comment.user_id === userId){
+                            story.comments = story.comments.filter(update => update.comment_id !== commentId);
```

Comment:
Do we have to convert userId to to "String" here before comparison ? It is possible that they send the userId as number from the front end. Since here we are using "===" If tthey send a number then this check may fail

Reply:
i think this was changed by vishal, and i agree with it, this ensures type safety. Also there is no conversion to string, the userId is a number in schema. So yes, the userid should always be passed as number, to be able to delete comment.

Reply:
Okay got it. So if they send a string we wont delete it.

Reply:
correct